### PR TITLE
Bug fix/keep mindepth on undecided condition

### PR DESCRIPTION
**Pull request by mchacki**
This Bug fix allows to place variable values into FilterConditions fo traversals.
In most cases the optimizer made the variables constant which is much easier to optimize, so we now have added explicit tests for non-const variables that require runtime information.

### Scope & Purpose

*(Can you describe what functional change your PR is trying to effect?)*

- [x] Bug-Fix for *devel-branch*
- [ ] Bug-Fix for a *released version* We need to back port it.
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added **Regression Tests** (Only for bug-fixes) 

Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10008/

### CODE REVIEW

**Comment by mchacki** [comment]
Enterprise companion:
https://github.com/arangodb/enterprise/pull/461

**Comment by mchacki** [comment]
Jenkins after JSLint fix:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10009/

File: arangod/Aql/TraversalConditionFinder.cpp [hunk 1]
```diff
@@ -747,7 +747,7 @@ bool TraversalConditionFinder::isTrueOnNull(AstNode* node, Variable const* pathV
     // Too complex, would require to figure out all
     // possible values for all others vars and play them through
     // Do not opt.
```
Comment (hkernbach) [inline arangod/Aql/TraversalConditionFinder.cpp:749]: Need to adjust comment?

File: arangod/Cluster/TraverserEngine.h [hunk 2]
```diff
@@ -85,7 +86,7 @@ class BaseEngine {
   virtual EngineType getType() const = 0;
 
   virtual bool produceVertices() const { return true; }
-
+
```
Comment (hkernbach) [inline arangod/Cluster/TraverserEngine.h:89]: Not necessarily needed :P

File: arangod/Aql/TraversalConditionFinder.cpp (hunk 1 above)
Comment (mchacki) [reply arangod/Aql/TraversalConditionFinder.cpp:749]: no this is still correct.

**Comment by jsteemann** [comment]
Code LGTM, but there has been a Jenkins test failure:
``` 
Test "shell_server_aql_4"
    [FAILED]  enterprise/tests/js/server/aql/aql-disjoint-smart-graph-traversal-generic-cluster.js

      "testOpenDiamondDfsLabelVariableForwarding_openDiamond_Graph_6shards_s0-A_s1-B_s2-C_s3-D_s4-E_s5-F" failed: ArangoError: AQL: missing node type in TypeNames (exception location: /work/ArangoDB/arangod/Aql/AstNode.cpp:946) (while optimizing plan)
    at ArangoStatement.execute (/work/ArangoDB/js/server/modules/@arangodb/arango-statement.js:84:16)
    at ArangoDatabase._query (/work/ArangoDB/js/server/modules/@arangodb/arango-database.js:102:45)
    at testOpenDiamondDfsLabelVariableForwarding (/work/ArangoDB/js/server/modules/@arangodb/aql-graph-traversal-generic-tests.js:1134:20)
    at Object.suite.<computed> (enterprise/tests/js/server/aql/aql-disjoint-smart-graph-traversal-generic-cluster.js:83:11)
    at Object.run (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:549:27)
    at Object.Run [as run] (/work/ArangoDB/js/common/modules/jsunity.js:276:24)
    at enterprise/tests/js/server/aql/aql-disjoint-smart-graph-traversal-generic-cluster.js:92:9
    at enterprise/tests/js/server/aql/aql-disjoint-smart-graph-traversal-generic-cluster.js:95:3
    at RunTest (/work/ArangoDB/js/common/modules/jsunity.js:390:12)
    at source (eval at <anonymous> (unknown source), <anonymous>:2:52) - test failed

      "testOpenDiamondBfsLabelVariableForwarding_openDiamond_Graph_6shards_s0-A_s1-B_s2-C_s3-D_s4-E_s5-F" failed: ArangoError: AQL: missing node type in TypeNames (exception location: /work/ArangoDB/arangod/Aql/AstNode.cpp:946) (while optimizing plan)
    at ArangoStatement.execute (/work/ArangoDB/js/server/modules/@arangodb/arango-statement.js:84:16)
    at ArangoDatabase._query (/work/ArangoDB/js/server/modules/@arangodb/arango-database.js:102:45)
    at testOpenDiamondBfsLabelVariableForwarding (/work/ArangoDB/js/server/modules/@arangodb/aql-graph-traversal-generic-tests.js:1367:20)
    at Object.suite.<computed> (enterprise/tests/js/server/aql/aql-disjoint-smart-graph-traversal-generic-cluster.js:83:11)
    at Object.run (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:549:27)
    at Object.Run [as run] (/work/ArangoDB/js/common/modules/jsunity.js:276:24)
    at enterprise/tests/js/server/aql/aql-disjoint-smart-graph-traversal-generic-cluster.js:92:9
    at enterprise/tests/js/server/aql/aql-disjoint-smart-graph-traversal-generic-cluster.js:95:3
    at RunTest (/work/ArangoDB/js/common/modules/jsunity.js:390:12)
    at source (eval at <anonymous> (unknown source), <anonymous>:2:52) - test failed

   Suites failed: 1 Tests Failed: 2
```

**Comment by hkernbach** [comment]
@jsteemann Yes, we're already investigating. Do not merge yet.

**Comment by mchacki** [comment]
Jenkins has been green:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10029/

Merged devel afterwards and rerun Jenkins:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10058/
This fails with the usual suspect, and a misconfiguration on a windows machine.

**Comment by mchacki** [comment]
Jenkins again:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10137/

**Comment by mchacki** [comment]
Jenkins bleu except the known one.

**Pull request merged**